src/framework/components/model/component.js

```diff
@@ -24,65 +26,121 @@ import { Component } from '../component.js';
  */
 class ModelComponent extends Component {
     /**
-     * Create a new ModelComponent instance.
-     *
-     * @param {ModelComponentSystem} system - The ComponentSystem that created this Component.
-     * @param {Entity} entity - The Entity that this Component is attached to.
+     * @type {string}
+     * @private
      */
-    constructor(system, entity) {
-        super(system, entity);
+    _type = 'asset';
 
-        this._type = 'asset';
-
-        /**
-         * @type {number}
-         * @private
-         */
-        this._asset = null;
-        /**
-         * @type {Model}
-         * @private
-         */
-        this._model = null;
-
-        this._mapping = {};
-
-        this._castShadows = true;
-        this._receiveShadows = true;
-
-        /**
-         * @type {Asset}
-         * @private
-         */
-        this._materialAsset = null;
-        this._material = system.defaultMaterial;
+    /**
+     * @type {Asset|number|null}
+     * @private
+     */
+    _asset = null;
 
-        this._castShadowsLightmap = true;
-        this._lightmapped = false;
-        this._lightmapSizeMultiplier = 1;
-        this._isStatic = false;
+    /**
+     * @type {Model|null}
+     * @private
+     */
+    _model = null;
 
-        this._layers = [LAYERID_WORLD]; // assign to the default world layer
-        this._batchGroupId = -1;
+    /* eslint-disable jsdoc/check-types */
+    /**
+     * @type {Object.<string, number>}
+     * @private
+     */
+    _mapping = {};
+    /* eslint-enable jsdoc/check-types */
```

Comment:
```suggestion
    /**
     * @type {Record<string, number>}
     * @private
     */
    _mapping = {};
```

It takes two little changes in `node_modules\@playcanvas\jsdoc-template\publish.js` to make this work without extra steps to ignore `eslint` and the resulting types would also look less verbose/strange:

```ts
    /**
     * A dictionary that holds material overrides for each mesh instance. Only applies to model
     * components of type 'asset'. The mapping contains pairs of mesh instance index - material
     * asset id.
     *
     * @type {Record<string, number>}
     */
    set mapping(arg: Record<string, number>);
    get mapping(): Record<string, number>;
```

vs.

```ts
    /**
     * A dictionary that holds material overrides for each mesh instance. Only applies to model
     * components of type 'asset'. The mapping contains pairs of mesh instance index - material
     * asset id.
     *
     * @type {Object.<string, number>}
     */
    set mapping(arg: {
        [x: string]: number;
    });
    get mapping(): {
        [x: string]: number;
    };
```

In `node_modules\@playcanvas\jsdoc-template\publish.js` you can replace this:

```js
    // Check for key-value (aka dictionary or index signature) types
    match = /^Object.<string,\s*(.*)>$/i.exec(name);
    if (match) {
        name = match[1];
```

with:

```js
    // Check for key-value (aka dictionary or index signature) types
    match = /^(Object|Record).<string,\s*(.*)>$/i.exec(name);
    if (match) {
        name = match[2]; // changed from [1] to [2] because extra group
```

Reply:
This is very interesting. However, there are actually several instances of this across the engine codebase now, so rather than just fix this one up here, I think I will open an issue on this and we can address is separately. Thanks for the tip!!